tests/test_util.py

```diff
@@ -0,0 +1,134 @@
+import os
+import types
+import unittest
+
+try:
+    from unittest import mock
+except ImportError:
+    # `Python 2` or lower than `Python 3.3` does not
+    # have the `unittest.mock` module built-in
+    import mock
+from pythonforandroid import util
+
+
+class TestUtil(unittest.TestCase):
+    @mock.patch("pythonforandroid.util.makedirs")
+    def test_ensure_dir(self, mock_makedirs):
+        util.ensure_dir("fake_directory")
+        mock_makedirs.assert_called_once_with("fake_directory")
+
+    @mock.patch("shutil.rmtree")
+    @mock.patch("pythonforandroid.util.mkdtemp")
+    def test_temp_directory(self, mock_mkdtemp, mock_shutil_rmtree):
+        mock_mkdtemp.return_value = "/temp/any_directory"
+        with util.temp_directory():
+            mock_mkdtemp.assert_called_once()
+        mock_shutil_rmtree.assert_called_once_with("/temp/any_directory")
```

Comment:
Minor remark: we could also have `mock_shutil_rmtree.assert_not_called()` while still inside the context manager